20240124_chwide-noise/chwide.py

```diff
@@ -0,0 +1,231 @@
+"""Module that produces simple noise maps for s4 Chile wide survey
+
+jcarron Jan 2024
+"""
+import glob
+import os
+import healpy as hp
+import numpy as np
+import numpy.random
+
+from lenspyx.utils_hp import Alm, almxfl, gauss_beam
+from lenspyx.remapping.utils_geom import Geom
+from lenspyx.utils import cli
+from scipy.interpolate import UnivariateSpline as spl
+from psutil import cpu_count
+
+from lenspyx.utils import timer
+from lenspyx import cachers
+
+UNSEEN = -1.6375e+30
+assert hp.UNSEEN == UNSEEN
+pix_min, npix = 14135296, 33839732
+
+# --- freqs on disk for alt1 chile wide ---
+freqs = [30, 40, 90, 150, 220, 280]
+XYs = ['II', 'IQ', 'IU', 'QQ', 'QU', 'UU']
+
+fns = glob.glob(os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide/*.fits')
+if len(fns) == 2 * len(freqs): # Working locally
+    print("Using files on " + os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide')
+    path2cov = os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide/cov_%03d.fits'  # in K^2, nside 128
+    path2hits = os.path.dirname(__file__) + '/inputs/alternative_1/chlat_cd_wide/hits_%03d.fits'
+else: # Most likely we are on NERSC
+    path2cov = '/global/cfs/cdirs/cmbs4/AoA/August2022/scaled_outputs/alternative_1/chlat_cd_wide/cov_%03d.fits' # in K^2, nside 128
+    path2hits = '/global/cfs/cdirs/cmbs4/AoA/August2022/scaled_outputs/alternative_1/chlat_cd_wide/hits_%03d.fits'
+
+nside = 128
+tht_min, tht_max = (64 / 180 * np.pi, 155. / 180 * np.pi)
+zbounds = np.cos([tht_max, tht_min])
+
+# --- PBDR 1/f noise parameters ---
+freqs_PBDR = [25, 40, 90, 150, 230, 280] # (! not the same as DC0)
```

Comment:
why not using the DC-0 figures?

Reply:
> + +# --- PBDR 1/f noise parameters --- +freqs_PBDR = [25, 40, 90, 150, 230, 280] # (! not the same as DC0) why not using the DC-0 figures?
> These values only enters to get a value for lknee and alpha at the similar DC-0 frequencies. Are there ‘officlal’ such numbers for DC0 ? I’d think it should not matter.

Official numbers are in the `s4sim` repository.
Now, unfortunately they do not make any Github release that would simplify versioning.
However, I stored the actual status of the `s4sim` repository when we ran DC-0:
https://github.com/CMB-S4/s4sim/blob/e20d9ab48f894888d78288ed9c0a7a659e0454bb/s4sim/pipeline_tools/hardware.py

So you could get noise properties from there. Maybe we are lucky and they are the same! (Joking, that would never happen)